consul_exporter.go

```diff
@@ -207,15 +227,35 @@ func (e *Exporter) setMetrics(services <-chan []*consul_api.ServiceEntry) {
 	}
 }
 
+func (e *Exporter) setKeyValues() {
+	kv := e.client.KV()
+
+	pairs, _, err := kv.List(e.kvPrefix, &consul_api.QueryOptions{})
+	if err != nil {
+		log.Errorf("Error fetching key/values: %s", err)
+	} else {
+		for _, pair := range pairs {
+			if e.kvFilter.MatchString(pair.Key) {
+				val, err := strconv.ParseFloat(string(pair.Value), 64)
+				if err == nil {
+					e.keyValues.WithLabelValues(pair.Key).Set(val)
+				}
+			}
+		}
+	}
+}
+
 func main() {
 	var (
 		listenAddress = flag.String("web.listen-address", ":9107", "Address to listen on for web interface and telemetry.")
 		metricsPath   = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
 		consulServer  = flag.String("consul.server", "localhost:8500", "HTTP API address of a Consul server or agent.")
+		kvPrefix      = flag.String("kv.prefix", "none", "Prefix from which to expose key/value pairs")
```

Comment:
Nit: periods at end of flag descriptions to be consistent with the other flags please.